contracts/test/CompoundGovernor.t.sol

```diff
@@ -736,3 +769,190 @@ contract CastVoteWithReasonAndParams is CompoundGovernorTest {
         governor.castVoteWithReasonAndParams(_proposalId, VOTE_TYPE_FRACTIONAL, "MyReason", _newParams);
     }
 }
+
+contract ProposalDeadline is CompoundGovernorTest {
+    function testFuzz_ProposalDeadlineCorrectWithEnumeratedId(uint256 _proposerIndex) public {
+        _proposerIndex = bound(_proposerIndex, 0, _majorDelegates.length - 1);
+        address _proposer = _majorDelegates[_proposerIndex];
+        uint256 _clockAtSubmit = governor.clock();
+        Proposal memory _proposal = _buildAnEmptyProposal();
+        uint256 _proposalId = _submitProposal(_proposer, _proposal);
+        uint256 _deadline = governor.proposalDeadline(_proposalId);
+        assertEq(_deadline, _clockAtSubmit + INITIAL_VOTING_DELAY + INITIAL_VOTING_PERIOD);
+    }
+
+    function testFuzz_ZeroReturnedIf_ProposalDeadlineCalledWithInvalidProposalId(
+        uint256 _invalidProposalId
+    ) public view{
+        uint256 _deadline = governor.proposalDeadline(_invalidProposalId);
+        assertEq(_deadline, 0);
+    }
+}
+        
+
+contract ProposalSnapshot is CompoundGovernorTest {
+    function testFuzz_ProposalSnapshotCorrectWithEnumeratedId(uint256 _proposerIndex) public {
+        _proposerIndex = bound(_proposerIndex, 0, _majorDelegates.length - 1);
+        address _proposer = _majorDelegates[_proposerIndex];
+        uint256 _clockAtSubmit = governor.clock();
+        Proposal memory _proposal = _buildAnEmptyProposal();
+        uint256 _proposalId = _submitProposal(_proposer, _proposal);
+        uint256 _snapShot = governor.proposalSnapshot(_proposalId);
+        assertEq(_snapShot, _clockAtSubmit + INITIAL_VOTING_DELAY);
+    }
+
+    function testFuzz_ZeroReturnedIf_ProposalSnapshotCalledWithInvalidProposalId(
+        uint256 _invalidProposalId
+    ) public view {
+        uint256 _snapShot = governor.proposalSnapshot(_invalidProposalId);
+        assertEq(_snapShot, 0);
+    }
+}
+
+contract ProposalEta is CompoundGovernorTest {
+    function testFuzz_ProposalEtaCorrectWithEnumeratedId(uint256 _proposerIndex) public {
+        _proposerIndex = bound(_proposerIndex, 0, _majorDelegates.length - 1);
```

Comment:
here and in rest of the lines we no longer need `_proposerIndex` to get a random proposer.

Reply:
Done: 9c1f15694fdc5e64ff58f639c67dd6f72d3e45dd